src/jaguar.toit

```diff
@@ -63,31 +67,56 @@ main args:
     // Don't keep trying to run malformed programs.
     manager.last = null
 
-  task::
-    identify id name address
-  server := http.Server --logger=logger
-  server.listen socket:: | request/http.Request writer/http.ResponseWriter |
-    device_id_header := request.headers.single DEVICE_ID_HEADER
-    sdk_version_header := request.headers.single SDK_VERSION_HEADER
-
-    // Validate device ID
-    if device_id_header != id.stringify:
-      logger.info "denied request, header: '$DEVICE_ID_HEADER' was '$device_id_header' not '$id'"
-      writer.write_headers 403 --message="Device has id '$id', jag is trying to talk to '$device_id_header'"
-
-    // Validate SDK version
-    else if sdk_version_header != vm_sdk_version:
-      logger.info "denied request, header: '$SDK_VERSION_HEADER' was '$sdk_version_header' not '$vm_sdk_version'"
-      writer.write_headers 406 --message="Device has $vm_sdk_version, jag has $sdk_version_header"
-
-    else if request.path == "/code" and request.method == "PUT":
-      install_program request.content_length request.body
-      writer.write
-        json.encode {"status": "success"}
-
-    else if request.path == "/ping" and request.method == "GET":
-      writer.write
-        json.encode {"status": "OK"}
+  while true:
+    failures := 0
+    attempts := 3
+    while failures < attempts:
+      exception = catch: run id name port
+      if not exception: continue
+      failures++
+      logger.warn "running Jaguar failed due to '$exception' ($failures/$attempts)"
+    backoff := Duration --s=5
+    logger.info "backing off for $backoff"
+    sleep backoff
+
+run id/uuid.Uuid name/string port/int:
+  broadcast_task := null
+  server_task := null
+  network/net.Interface? := null
+  error := null
+
+  try:
+    network = net.open
+    socket/tcp.ServerSocket := network.tcp_listen port
+    address := "http://$network.address:$socket.local_address.port"
+    logger.info "running Jaguar device '$name' (id: '$id') on '$address'"
+
+    // We run two tasks concurrently: One broadcasts the device identity
```

Comment:
This appears to be generally useful. Maybe we can factor this out and let other's benefit from this support for running multiple tasks that start/stop as one?